isomerpages/isomercms-frontend
# refactor: abstract menu dropdown button into a common component

## Problem



The current menu dropdown button is used in the Folders view and in the Workspace view when there are no pages and folders in the site. However, the component is useful and can be generalised such that other components can reuse it (such as the site dashboard).

## Solution



**Breaking Changes**




**Improvements**:

- The existing menu dropdown button has been abstracted out as a common component and introduced the ability to customise the component in terms of:
    - The button variant
    - The text of the main button (plus other props)
    - The items within the context menu
- The storybook stories have also been added for the menu dropdown button

**Caveats**:

- The code to use the menu dropdown button for both Workspace and Folders view violates the DRY principle. This is intended, as per Design, the menu dropdown button for the Workspace empty area view will be replaced with two separate buttons instead.

## Tests




1. Navigate to any site's workspace and click into one of the folders.
2. Verify that the menu dropdown button to create pages and folders have not changed.

## Deploy Notes




*None*
